File: python3/is_subsequence.py
"""
Space   : O(1)
Time    : O(n)
"""

class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        m, n = len(s), len(t)

        i = 0
        for j in range(n):
            if i == m:
                return True

            if s[i] == t[j]:
                i += 1

        if i == m:
            return True
        
        return False
        
        

File: golang/is_subsequence.go
package main

// Space	: O(1)
// Time		: O(n)

func isSubsequence(s string, t string) bool {
	m, n := len(s), len(t)

	i := 0
	for j := 0; j < n; j++ {
		if i == m {
			return true
		}

		if s[i] == t[j] {
			i++
		}
	}

	if i == m {
		return true
	}

	return false
}
